### DISCUSSION TITLE
Data source with manual pagination configuration

### DISCUSSION THREAD

**Discussion by micebrain** [Budibase/budibase#8554] [Ideas]
Hi,
I am able to implement manual pagination for data source with custom SQL queries. However this comes with a few problems that on has to keep in mind:
- the "Filter" in data source cannot work properly since the filter is applied on the result from the SQL query with the limit clause
- the "Limit" in data source should be set to be the same manual limit to avoid confusion of missing rows

Is it possible that the data source expose e.g. the limit property so that if can be bindable to the query?

Or it would be nice to allow ordered "key columns" to be configured on data source so that it can be used to support the pagination. 

Thanks for all the great work!

**Comment by melohagan** [comment]
Hey @micebrain 

You may find this guide useful: https://github.com/Budibase/budibase/discussions/7313#discussioncomment-3430353

It shows how you can bind to the LIMIT and OFFSET of a SQL query.
